numpy/f2py/src/fortranobject.c

```diff
@@ -19,15 +19,21 @@ extern "C" {
 int
 F2PyDict_SetItemString(PyObject *dict, char *name, PyObject *obj)
 {
+    int retval = -1;
     if (obj == NULL) {
         fprintf(stderr, "Error loading %s\n", name);
         if (PyErr_Occurred()) {
             PyErr_Print();
             PyErr_Clear();
         }
+        Py_DECREF(obj);
         return -1;
     }
-    return PyDict_SetItemString(dict, name, obj);
+    else {
+        retval = PyDict_SetItemString(dict, name, obj);
+        Py_XDECREF(obj);  // Cannot be NULL, is OK
```

Comment:
You mean DECREF :), and I would skip the comment, it is pretty clear, especially if you use it just above.

Reply:
Oops, mixed them up >.<; great catch!